### PR TITLE
eth-typing 5.0.1 :snowflake: 

### PR DESCRIPTION
**Pull request by AndrewOsip**
eth-typing 5.0.1 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-5939]() 
- [Upstream repository](https://github.com/ethereum/eth-typing)

[PKG-5939]: https://anaconda.atlassian.net/browse/PKG-5939?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by tombenes — APPROVED** [review]

**Review by tombenes — CHANGES_REQUESTED** [review]
This package has target channel snowflake, so you need to create `abs.yaml` file in root dir with:
```
upload_channels:
  - sfe1ed40
````

**Review by tombenes — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 177e2070da9bf557fe0fd46ee467a7be2d0b6476aa4dc18680603e7da1fc5690
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name.replace('-', '_') }}-{{ version }}.tar.gz
+  sha256: 83debf88c9df286db43bb7374974681ebcc9f048fac81be2548dbc549a3203c0
 
 build:
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
   number: 0
```
Comment (lorepirri) [inline recipe/meta.yaml:14]:
```suggestion
  number: 0
  skip: True  # [py<38]
```
https://github.com/ethereum/eth-typing/blob/432fe9eeb945416d8704d69c321235e0b6bd248d/setup.py#L52

File: recipe/meta.yaml [hunk 2]
```diff
@@ -34,8 +34,12 @@ about:
...
+  license_family: MIT
   license_file: LICENSE
+  doc_url: https://github.com/ethereum/eth-typing/blob/main/README.md
+  dev_url: https://github.com/ethereum/eth-typing
+  description: Common type annotations for ethereum python packages
 
 extra:
   recipe-maintainers:
-    - step21
+    - snegireff
```
Comment (lorepirri) [inline recipe/meta.yaml]: usually, we do not update maintainers unless it's a new feedstock or a total rewrite

**Review by lorepirri — APPROVED** [review]
Thanks for working on this!

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
   number: 0
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv
 
 requirements:
   host:
+    - python
     - pip
-    - python >=3.5,<4.0
+    - wheel
```
Comment (lorepirri) [inline recipe/meta.yaml:21]:
```suggestion
    - wheel
    - setuptools
```

@Cansisti @AndrewOsip this is needed

**Review by lorepirri — APPROVED** [review]
LGMT, Thanks for improving it

**Review by skupr-anaconda — APPROVED** [review]
Thanks!

**Pull request merged**